src/sparseml/onnx/benchmark/info.py

```diff
@@ -0,0 +1,599 @@
+            raise ValueError(f"Provider {ort_provider} not installed.")
+
+        self._model_runner = ORTModelRunner(
+            self._model,
+            batch_size=batch_size,
+            providers=[ort_provider],
+            **framework_args,
+        )
+
+        self._inference_provider = inference_providers[0]
+        self._provider = provider
+        self._device = device
+        self._framework_args = framework_args
+
+        self._batch_size = batch_size
+        self._iterations = iterations
+        self._warmup_iterations = warmup_iterations
+
+    def run_batch(
+        self, batch: Union[Dict[str, Any], Tuple[Dict[str, Any], Any]], *args, **kwargs
+    ) -> BatchBenchmarkResult:
+        """
+        Runs a benchmark on a given batch.
+
+        :param batch: the batch to benchmark
+        :param args: additional arguments to pass to the framework
+        :param kwargs: additional arguments to pass to the framework
+        """
+        # Handles case where batch consists of a tuple of input/labels
+        if isinstance(batch, tuple):
+            batch = batch[0]
+        _, batch_time = self._model_runner.batch_forward(batch, *args, **kwargs)
+        return BatchBenchmarkResult.from_result(batch_time, self.batch_size)
+
+    @property
+    def framework(self) -> Framework:
+        """
+        :return: the framework
+        """
+        return Framework.onnx
+
+    @property
+    def framework_info(self) -> FrameworkInfo:
+        """
+        :return: the framework info
+        """
+        return self._framework_info
+
+    @property
+    def batch_size(self) -> int:
+        """
+        :return: the batch size
+        """
+        return self._batch_size
+
+    @property
+    def warmup_iterations(self) -> int:
+        """
+        :return: the warmup iterations
+        """
+        return self._warmup_iterations
+
+    @property
+    def iterations(self) -> int:
+        """
+        :return: the number of iterations
+        """
+        return self._iterations
+
+    @property
+    def inference_provider(self) -> FrameworkInferenceProviderInfo:
+        """
+        :return: the inference provider
+        """
+        return self._inference_provider
+
+    @property
+    def package_versions(self) -> Dict[str, str]:
+        """
+        :return: the package versions
+        """
+        return self._package_versions
+
+    @property
+    def framework_args(self) -> Dict[str, Any]:
+        """
+        :return: the framework args
+        """
+        return self._framework_args
+
+    @property
+    def device(self) -> str:
+        """
+        :return: the device
+        """
+        return self._device
+
+    @property
+    def model(self) -> ModelProto:
+        """
+        :return: the model as an ONNX ModelProto
+        """
+        return self._model
+
+
+class ORTCudaBenchmarkRunner(ORTBenchmarkRunner):
```

Comment:
I'm worried about this scaling as well across all of the different providers we may want to support in the future for ORT. Is there anyway to make this more generic and set a provider or list of providers arg and then run checks based on that?